### PR TITLE
make ImageDataGenerator generator repeatable

### PR DESCRIPTION
**Pull request by wassname**
We want ImageDataGenerator to be repeatable so we can 1) have repeatable research 2) so we can transform a mask and image in parallel (like in the keras docs).

In PR #3751 I made it mostly repeatable but I recently ran more tests and found problems. It's not repeatable when using flow_from_directory, and it's not repeatable if the mask and image have different number of channels. 

This PR fixes that by adding `np.random.seed(seed)` where needed. 

I have two unit tests I can add if desired?

### CODE REVIEW

**Comment by wassname** [comment]
Could someone give this a review? I found a [issue in keras_retinanet](https://github.com/fizyr/keras-retinanet/issues/150) that would be fixed by this, so it's still an issue. It wont take long I just added `np.random.seed(seed)` where needed.

**Comment by ahundt** [comment]
I think these changes would need a unit test that demonstrates the problem that is fixed to have any chance of being merged.

**Comment by de-vri-es** [comment]
I don't think resetting the random number generator to the seed for every invocation of the RNG is a good idea. You're effectively reducing it to generate one random number and repeating it a lot of times.

The proper way to solve the problem (at-least for keras-retinanet) I think is to split the generation and usage of random transformations into separate steps so that the random transformation can be used multiple times on different things (like the image and the bounding boxes).

**Comment by wassname** [comment]
@de-vri-es That's true, my fix is not ideal, and your proposal would be much cleaner. But it would take much longer and it seems like no one is keen to spend much time on this part of keras. Which is fair enough, perhaps it could be split into a separate package to shift the burden of keras maintainers.

@ahundt In the original posts I offered unit tests, but I deleted them since there was no interest. I could look into writing them again, but would you even want this PR if I did, or would it be a waste of my time?

**Comment by ahundt** [comment]
@wassname Yeah probably wouldn't be merged considering it repeats values. 

I think a solution like the one @de-vri-es suggested would be necessary or something like SegDataGenerator as I mentioned in the [jaccard distance PR](https://github.com/farizrahman4u/keras-contrib/pull/152#discussion_r148966647). 

For other readers' reference: [SegDataGenerator](https://github.com/aurora95/Keras-FCN/blob/master/utils/SegDataGenerator.py#L291) from [Keras-FCN](https://github.com/aurora95/Keras-FCN) transforms both an image and a mask. A usage example is available in the Keras-FCN's [train.py]( https://github.com/aurora95/Keras-FCN/blob/master/train.py#L117) file.

**Comment by wassname** [comment]
I agree this was just a quick fix, sounds like we all want to leave it until there is a better solution. 

Yeah SegDataGenerator is pretty nice for segmentation, I've used it and found it easy to edit too. Might be worth adapting to keras_retinanet too, thanks for the suggestion.
